app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,101 @@
+class PasswordResetsController < ApplicationController
+  def new
+  end
+
+  def create
+    if params[:username].blank?
+      flash.now[:alert] = 'Please enter a username'
+      render 'new', status: :unprocessable_entity
+      return
+    end
+
+    begin
+      user = User.find(escaped_username)
+
+      if user.nil?
+        user_not_found
+      else
+        PasswordResetMailer.password_reset(user).deliver
+        flash.now[:notice] = 'Your password reset email has been sent'
+        @status = :ok
+      end
+    rescue Net::HTTPServerException => e
```

Comment:
This rescue, based on my code reading, only applies to line 13, the User.find line. I think it's be better to shift this to only be around that line, and have the user.nil and password_reset call be outside of the rescue block, that way we reduce any confusion on if this rescue applies to that code.